wordpress-mobile/WordPress-FluxC-Android
# [Detekt Baseline Warnings] FluxC Module - Resolve/Suppress Style Warnings

Parent #2479
Partially Closes: #2485

This PR resolves/suppresses all `style` related warnings for the `fluxc` module:

1. `70` x DataClassShouldBeImmutable (Resolve: 8a96a86fb723285d8a056231ebad6096231aadd9 + Suppress: b01843467e6c2dcf73975233262c1931e9e50a90)
2. `1` x ExplicitItLambdaParameter (Resolve: 82aaf44051acf0d8041b0cf75f2435da7ad57fcb)
3. `10` x ForbiddenComment (Suppress: 3b61e96d8f04cbf4f7fdbdee65780be80e08ff2e + 8b1b57004ff23f1e160243b2043ae4283e3f204c)
4. `18` x MagicNumber (Resolve: 1eb75cb183e41fc9aca4669b066df2f41c36e2a1 + fcfcfeeb4f32f6b9b2dfa87a9e33b6fa0a779f94)
5. `15` x MaxLineLength (Resolve: 22addfcc6ae947cdc8ee02227a959215af8e8638 + Suppress: d3108042bcc93efc706ce72be39d16b89840f0ea)
6. `10` x ReturnCount (Resolve: 3be3c5327d6832ce09b94b5fc0bd93a77e9a6873 + Suppress: eae42a718c522e586e2517c5d7a9d9b35b06682c)
7. `1` x UnnecessaryAbstractClass (Resolve: 62f956576c0e08d45f0d4a459f9dc16cbc1fb3a2)
8. `4` x UnusedPrivateMember (Resolve: 11333dd77f23817ff1d6723cd7b887238f44c98c + f6e3c74df7a7f89b001a787d3bfde3d0d2917dab + Suppress: 6bc859216a5b15094a5c516cb4d571a863e3ad9d)
9. `1` x UseCheckOrError (Resolve: dd8ac78a067adcc1e7a94ae0174927b0d4568ade)
10. `2` x UtilityClassWithPublicConstructor (Resolve: 8e683387721b1195b94e9421d1c46e9dfe0f0bb5)

PS.1: I am randomly adding one engineer from the WPAndroid team and one from the WCAndroid team as the main reviewers, that is, in addition to the @wordpress-mobile/owl-team team itself, since I just want someone from these teams to sign-off on that change as well (or additionally to the 🦉 team).

PS.2: I recommend reviewing this PR commit-by-commit.

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough (especially the `detekt` check).
- However, if you really want to be thorough, you could smoke test the `example` and/or the `WPAndroid/WCAndroid` apps to verify that everything works as expected.

PS: Please ignore the failing `connected-tests` check as they are unrelated to this PR. This check has been failing consistency for some time now due to tests being flaky or needing to be updated/fixed.
